src/Adapter/ContextStateManager.php

```diff
@@ -136,8 +136,10 @@ public function setCurrency(?Currency $currency): self
      */
     public function setLanguage(?Language $language): self
     {
-        $this->saveContextField('language');
-        $this->context->language = $language;
+        if (null !== $language->id) {
```

Comment:
before this modif, If you called setLanguage(null), the context->language was set to null, now this call will have no effect. Isn't it a BC break ?

Reply:
You're right, I'm not even checking if `$language` is `null` here 🤔
But I can't find any place where we call this method with `null`. Maybe @jolelievre can give us some insights about why we allow `null` here?

Reply:
I don't see any use case for now indeed I think it was a safety because we are not always sure if the injected data is really present or not And in some case we might also want to force the context value to be `null` (To force a behaviour to refill it or to avoid using a contextual value..)

I think it would be preferable not to change this, ideally, especially since all the other setters don't have this behaviour Why did you need this modification by the way?

Reply:
@jolelievre I need this modification because when a language has been deleted, you can create the object `Language` with this `id` of the deleted language, you'll get the object (not `null`), but the `id` will be null. Then, when you try to get a `Currency` object (for instance), you'll have an exception because [here](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/classes/Currency.php#L251-L253) `$idLang` will be null and [here](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/classes/Currency.php#L265) it will throw an `out of array` exception.

Reply:
Then I'd say the responsibility lies in the hand of the service that use the ContextStateManager, if it's an handler for example then it's the handler which should check that:
- the Language associated to order exists
- if it doesn't fallback to default language
- finally set it in the ContextStateManager